lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -90,6 +99,18 @@ void HostManager::removeNetwork(llvm::StringRef networkName) {
   if (networkIterator == networks_.end()) {
     return;
   }
+  auto &nodes = networkIterator->second.nodes;
+  for (auto &node : nodes) {
+    std::promise<llvm::Error> removeNetwork;
```

Comment:
This seems to cause issues for Windows (right @ayermolo?) so let's just pass an llvm::Error in by ref (can be initialized to success).

Reply:
Yep. Relevant discussion: https://github.com/pytorch/glow/pull/2477

Reply:
What part causes issues? Using an error in a promise?

Reply:
Two issues.
1) using llvm error in promise results in compilation error in visual studio 2017 with llvm 7.0.0
2) There is run time stack corruption when llvm gets executed.

The conclusion seems to be templated std::promise windows implementation has issues. @nickgg  proposed a solution that seems to work.

Reply:
Thanks, I'll use promise<void>.

Reply:
nit: you can use backticks and do `std::promise<void>` 😃

Reply:
I think it should be good to go now.